woocommerce/woocommerce
# Adding WooProductFieldItem slotfill

### All Submissions:






### Changes proposed in this Pull Request:



This PR introduces the WooProductFieldItem slot fill, which will be used to inject fields into the various product sections. 

Closes #36014  .

### How to test the changes in this Pull Request:



1. Checkout branch
2. Enable the `new-product-management-experience` feature flag. 
3. Include a test slot anywhere within the application JS. Something like this:
```js
/**
 * External dependencies
 */
import { __experimentalWooProductFieldItem as WooProductFieldItem } from '@woocommerce/components';
import { TextControl } from '@wordpress/components';
import { registerPlugin } from '@wordpress/plugins';

const DetailsNameField = () => {
	return (
		<TextControl
			label="Name"
			name={ `product-mvp-name` }
			placeholder="e.g. 12 oz Coffee Mug"
			value="Test Name"
			onChange={ () => console.debug( 'Changed!' ) }
		/>
	);
};

registerPlugin( 'wc-admin-product-editor-name', {
	// @ts-expect-error 'scope' does exist. @types/wordpress__plugins is outdated.
	scope: 'woocommerce-admin',
	render: () => (
		<WooProductFieldItem
			id="test-name-field"
			section="details"
			pluginId="test-plugin"
		>
			{ () => {
				return <DetailsNameField />;
			} }
		</WooProductFieldItem>
	),
} );
```
4. Navigate to Products -> Add New (MVP)
5. You should see the name field appear at the bottom of the details section:

![image](https://user-images.githubusercontent.com/444632/210911893-8340722d-d329-43ce-92e6-940fbbc9e38a.png)




### Other information:




### FOR PR REVIEWER ONLY:
